packages/app-desktop/gui/MainScreen/commands/toggleEditorFullScreen.ts

```diff
@@ -0,0 +1,64 @@
+import CommandService, {
+	CommandContext,
+	CommandDeclaration,
+	CommandRuntime,
+} from '@joplin/lib/services/CommandService';
+import { _ } from '@joplin/lib/locale';
+import setLayoutItemProps from '../../ResizableLayout/utils/setLayoutItemProps';
+import layoutItemProp from '../../ResizableLayout/utils/layoutItemProp';
+import { AppState } from '../../../app.reducer';
+
+// Record the current state of the sidebar and note list
+const lastUserLayoutStatus = {
+	isNoteListVisible: true,
+	isSidebarVisible: true,
+};
+
+export const declaration: CommandDeclaration = {
+	name: 'toggleEditorFullScreen',
+	label: () => _('Toggle fullscreen'),
+	iconName: 'fas fa-expand',
+};
+
+export const runtime = (): CommandRuntime => {
+	return {
+		execute: async (context: CommandContext) => {
+			const layout = (context.state as AppState).mainLayout;
+
+			const isNoteListVisible = layoutItemProp(layout, 'noteList', 'visible');
+			const isSidebarVisible = layoutItemProp(layout, 'sideBar', 'visible');
+			let isEditorFullScreen = layoutItemProp(layout, 'editor', 'isFullScreen');
+
+			if (isEditorFullScreen === undefined) {
+				isEditorFullScreen = !isNoteListVisible && !isSidebarVisible;
+			}
+
+			let newLayout = layout;
+
+			if (isEditorFullScreen) {
+				newLayout = await CommandService.instance().execute('toggleSideBar', lastUserLayoutStatus.isSidebarVisible);
+				newLayout = await CommandService.instance().execute('toggleNoteList', lastUserLayoutStatus.isNoteListVisible);
+			} else {
+				lastUserLayoutStatus.isSidebarVisible = isSidebarVisible;
+				lastUserLayoutStatus.isNoteListVisible = isNoteListVisible;
+
+				newLayout = await CommandService.instance().execute('toggleSideBar', false);
+				newLayout = await CommandService.instance().execute('toggleNoteList', false);
+			}
+
+			newLayout = setLayoutItemProps(newLayout, 'editor', {
+				isFullScreen: !isEditorFullScreen,
```

Comment:
Also here you set this "isFullScreen" property but it's not even defined on the LayoutItem interface. It works because the code your calling is muting type checking.

Reply:
Not sure what's the best approach for all this but as of now it's a strange hybrid solution that looks like functional code but actually based on global state. I don't know what to suggest I'm afraid, I just don't think it's worth spending more time on it. How about creating a plugin, wouldn't that work?

Reply:
Thanks for the review. I agree maybe it is not the best approach to achieve this feature  as I don't have much time to study the whole project and it's design. But personally speaking, this feature is very convenient and wish there will be an official achievement. Anyway, I may keep this feature in my personal version and close the full request in later time.